### PR TITLE
smart join views

### PR DESCRIPTION
**Pull request by ObiWahn**
### Scope & Purpose
We want to allow smart joins for views if the collections that form the view are sharded in a way that smart joins are possible.

### Checklist
- [X] Strictly *new functionality* new feature devel/enterprise only
- [X] The behavior change can be verified via automatic tests
- [X] Added new **integration tests** in shell_server (enterprise/cluster)
- [x] Added a *Changelog Entry* (referencing the corresponding public or internal issue number)
- [x] Added entry to *Release Notes* 
- [x] Updated section in the *Manual* 

Enterprise PR: https://github.com/arangodb/enterprise/pull/347
Docs: https://github.com/arangodb/docs/pull/191

test: http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/6854/

### CODE REVIEW

**Comment by ObiWahn** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-linux/8568/ - blue except for unrelated fail in hotbackup

File: arangod/Aql/IResearchViewNode.h [hunk 1]
```diff
@@ -219,6 +236,11 @@ class IResearchViewNode final : public arangodb::aql::ExecutionNode {
 
   /// @brief IResearchViewNode options
   Options _options;
+
+  /// @brief prototype collection for enabling smart joins
+  aql::Collection const* _prototypeCollection;
```
Comment (gnusi) [inline arangod/Aql/IResearchViewNode.h]: Please initialize these variables

Comment (gnusi) [inline arangod/Aql/IResearchViewNode.h]: BTW what is the purpose of having these variables as `IResearchViewNode` members?

**Comment by ObiWahn** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/6951/

**Comment by ObiWahn** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/6955/

**Comment by ObiWahn** [comment]
https://172.16.10.101/view/PR/job/arangodb-matrix-pr/6959/

**Comment by jsteemann** [comment]
Tests blue

**Comment by ObiWahn** [comment]
https://172.16.10.101/view/PR/job/arangodb-matrix-pr/6968/

**Pull request merged**